cmd/gomobile/env.go

```diff
@@ -311,10 +322,18 @@ func archClang(goarch string) string {
 		return "i386"
 	case "amd64":
 		return "x86_64"
-	case "macos64":
+	case "macos-amd64":
 		return "x86_64"
-	case "uikitMac64":
+	case "macos-arm64":
+		return "arm64"
+	case "sim-arm64":
+		return "arm64"
+	case "sim-amd64":
 		return "x86_64"
```

Comment:
This is very nitpicky but..
```suggestion
	case "sim-amd64":
		return "x86_64"
	case "sim-arm64":
		return "arm64"
```
for consistency